src/Parser/Priest/Shadow/Modules/Core/Haste.js

```diff
@@ -0,0 +1,80 @@
+import CoreHaste from 'Parser/Core/Modules/Haste';
+import SPELLS from 'common/SPELLS';
+
+const debug = false;
+
+const VOIDFORM_HASTE_PER_STACK = 0.01;
+const LINGERING_INSANITY_HASTE_PER_STACK = 0.01;
+
+
+class Haste extends CoreHaste {
```

Comment:
Maybe I'm missing something, but it's not clear to me why you need to extend CoreHaste. Why not just register Lingering Insanity and Voidform buffs in the core Haste module and let it handle things?

It looks like some manner of special mechanics happening here, I just don't understand what they are.

Reply:
I haven't successfully managed to get the Core Haste module correct with just adding the lingering and voidform buffs as this results in way to little haste overall. Most likely related to lingering insanity.

There is also a gap inbetween when you lose voidform and before lingering insanity is applied, due to not actually gaining a buff, but rather updating when you start to lose stacks.